x-pack/plugins/security_solution/public/plugin.tsx

```diff
@@ -355,11 +364,17 @@ export class Plugin implements IPlugin<PluginSetup, PluginStart, SetupPlugins, S
     return {};
   }
 
-  private async downloadAssets() {
-    const [{ renderApp }, { composeLibs }] = await Promise.all([
-      import('./app'),
-      import('./common/lib/compose/kibana_compose'),
-    ]);
+  private async downloadAssets(): Promise<{
+    composeLibs: (coreStart: CoreStart) => AppFrontendLibs;
+    renderApp: (props: RenderAppProps) => () => void;
+  }> {
+    /**
+     * The specially formatted comment in the `import` expression causes the corresponding webpack chunk to be named. This aids us in debugging chunk size issues.
+     * See https://webpack.js.org/api/module-methods/#magic-comments
+     */
+    const { renderApp, composeLibs } = await import(
+      /* webpackChunkName: "lazyPluginDependencies" */ './lazy_plugin_dependencies'
```

Comment:
I get that the major win here is from consolidating the multiple async imports (which each had some overlap) into one, but I'm curious: does the naming of these chunks have any impact on the build? That is, if each of these async imports had a different name, would would it result in multiple identical chunks, or is webpack smart enough to dedupe those since they're from the same file?

Reply:
I'll give this a try and report back. Also this PR loads the `createStore` and `createInitialState` functions dynamically now. That contributes to the decrease in initial bundle size.

Reply:
I gave one of the copies a different name:
![image](https://user-images.githubusercontent.com/35559/94732169-e7aca980-0333-11eb-9cf6-9dccb15de4b3.png)

The `stats.json` looks pretty much the same. the odd name doesn't show up. 
![image](https://user-images.githubusercontent.com/35559/94732313-19257500-0334-11eb-907d-75d46bfab922.png)

I would try again with another one of the copies renamed, but it freezes my computer for ~20 minutes :P

Reply:
Good to know, thanks for indulging me! 👍